doc/tutorials/using_ubuntu_as_sos.rst

```diff
@@ -0,0 +1,272 @@
+        the ``root=`` parameter (or use the device node directly).
+
+     .. note::
+        You will also need to adjust the kernel name if you used a different
+        RPM file as the source of your Service OS kernel.
+
+   * Update Grub on your system
+
+     .. code-block:: console
+
+        sudo update-grub
+
+     At this point, you need to modify ``/boot/grub/grub.cfg`` file manually to
+     enable the timeout so that the system has an opportunity to show you the
+     grub menu. (Without this the grub choice menu won't display.)
+
+     .. code-block:: none
+
+        #set timeout_style=hidden
+        set timeout = 10
+
+#. Reboot the system
+   
+   Reboot system. You should see the Grub menu with the new “ACRN ubuntu SOS”
+   entry. Select it and proceed to booting the platform. The system will start
+   the Ubuntu Desktop and you can now log in (as before).
+
+   To check if the hypervisor is effectively running, check ``dmesg``. The
+   typical output of a successful installation will look like this:
+
+   .. code-block:: none
+
+      dmesg | grep ACRN
+      [    0.000000] Hypervisor detected: ACRN
+      [    0.862942] ACRN HVLog: acrn_hvlog_init
+
+Prepare the User OS (UOS)
+*************************
+
+We are using a User OS based on `Clear Linux`_.
+
+* Download the Clear Linux image from `<https://download.clearlinux.org>`_
+
+  .. code-block:: console
+
+     cd ~
+     wget https://download.clearlinux.org/releases/22780/clear/clear-22780-kvm.img.xz
+     unxz clear-22780-kvm.img.xz
+
+* Download the Production Kenrel (PK) kernel
+
+  .. code-block:: console
+
+     wget https://download.clearlinux.org/releases/22780/clear/x86_64/os/Packages/linux-pk414-standard-4.14.47-44.x86_64.rpm
+     rpm2cpio linux-pk414-standard-4.14.47-44.x86_64.rpm | cpio -idmv
+
+* Update the UOS kernel modules
+
+  .. code-block:: console
+
+     sudo losetup -f -P --show /root/clear-22789-kvm.img
+     sudo mount /dev/loop0p3 /mnt
+     sudo cp -r /root/usr/lib/modules/4.14.47-44.pk414-standard /mnt/lib/modules/
+     sudo cp -r /root/usr/lib/kernel /lib/modules/
+     sudo umount /mnt
+     sync
+
+  If you encounter a permission issue, follow these steps:
+
+  .. code-block:: console
+
+     sudo chmod 777 /dev/acrn_vhm
+
+* One additional package is needed
+
+  .. code-block:: console
+
+     sudo apt-get instal iasl
+     sudo cp /usr/bin/iasl /usr/sbin/iasl
+
+* Adjust ``launch_uos.sh``
+ 
+  You need to adjust the ``/usr/share/acrn/samples/nuc/launch_uos.sh`` script
+  to match your installation. These are the couple of lines you need to modify:
+
+  .. code-block:: none
+
+     -s 3,virtio-blk,/root/clear-22780-kvm.img
+     -k /lib/modules/kernel/org.clearlinux.pk414-standard.4.14.47-44
+
+Start the User OS (UOS)
+***********************
+
+You are now all set to start the User OS (UOS)
+
+.. code-block:: console
+
+   sudo /usr/share/acrn/samples/nuc/launch_uos.sh
+
+**Congratulations**, you are now watching the User OS booting up!
```

Comment:
How about saying,

    With that, you've successfully booted the Clear Linux User OS (UOS) with an Ubuntu Service OS (SOS).